src/PIL/Image.py

```diff
@@ -1272,6 +1272,98 @@ def getdata(self, band=None):
             return self.im.getband(band)
         return self.im  # could be abused
 
+    def getdominantcolors(self, numcolors=3, maxiter=50, threshold=1):
+        """
+        Returns a list of dominant colors in an image using k-means
+        clustering.
+
+        :param numcolors: Number of dominant colors to search for.
+        The default number is 3.
+        :param maxiter: Maximum number of iterations to run the
+        algorithm.  The default limit is 50.
+        :param threshold: Early stopping condition for the algorithm.
+        Higher values correspond with increased color differences. The
+        default is set to 1 (corresponding to 1 pixel difference).
+        :returns: An unsorted list of (pixel) values.
+        """
+
+        if self.mode in ("F", "I", "L", "P"):
+            channels = 1
+        elif self.mode in ("RGB", "YCbCr", "LAB", "HSV"):
+            channels = 3
+        elif self.mode in ("RGBA", "CMYK"):
+            channels = 4
+        else:
+            tb = sys.exc_info()[2]
+            raise ValueError("Unsupported image mode").with_traceback(tb)
+
+        def euclidean(p1, p2):
+            if channels == 1:
+                return (p1 - p2) ** 2
+            return sum([(p1[i] - p2[i]) ** 2 for i in range(channels)])
+
+        w, h = self.size
+
+        pixels_and_counts = []
+        for count, color in self.getcolors(w * h):
```

Comment:
```suggestion
        pixels_and_counts = []
        for count, color in self.getcolors(self.width * self.height):
```
Just because you don't use `w` or `h` anywhere else, this looks simpler to me.